File: rpg-client-java/src/main/Nivel4.java
package main;

import static niveles.Nivel4.*;
import static utilidades.Acciones.*;

/*
Repite el nivel anterior pero utilizando funciones
*/

public class Nivel4 {

	public static void main(String[] args) {
		new Nivel4().comienzo();
	}

	public void comienzo() {

		iniciar();

		moverALechuga();
		recolectar();

		finalizar();
	}

	public void moverALechuga() {

		girarIzquierda();
		moverIzquierda();
		moverIzquierda();
		moverIzquierda();
		moverIzquierda();
		moverIzquierda();
		moverIzquierda();
		moverIzquierda();
		moverIzquierda();
		moverIzquierda();
		moverIzquierda();
		moverIzquierda();
		moverIzquierda();
		moverIzquierda();

		girarArriba();
		moverArriba();
		moverArriba();
		moverArriba();
		moverArriba();
		moverArriba();
		moverArriba();
		moverArriba();
		moverArriba();
		moverArriba();
		moverArriba();
		moverArriba();
		moverArriba();
		moverArriba();

		girarIzquierda();
		moverIzquierda();
		moverIzquierda();
		moverIzquierda();
		moverIzquierda();
		moverIzquierda();
		moverIzquierda();
		moverIzquierda();
		moverIzquierda();
		moverIzquierda();

		girarArriba();

		moverArriba();
		moverArriba();
		moverArriba();
	}

}

File: rpg-client-javascript/src/nivel4.js
const {
  iniciar,
  finalizar
} = require('./niveles/nivel4');

const {
  moverArriba,
  moverIzquierda,
  girarArriba,
  recolectar,
  girarIzquierda
} = require('./utilidades/acciones');

iniciar();

/*
  Recolecta la lechuga del huerto. Utiliza los siguientes verbos:
  - moverArriba
  - girarArriba
  - moverIzquierda
  - girarIzquierda
  - recolectar
/* */

function moverALechuga() {

  girarIzquierda();
  moverIzquierda();
  moverIzquierda();
  moverIzquierda();
  moverIzquierda(); 
  moverIzquierda(); 
  moverIzquierda(); 
  moverIzquierda(); 
  moverIzquierda(); 
  moverIzquierda(); 
  moverIzquierda(); 
  moverIzquierda(); 
  moverIzquierda(); 
  moverIzquierda(); 

  girarArriba();
  moverArriba();
  moverArriba();
  moverArriba();
  moverArriba();
  moverArriba();
  moverArriba();
  moverArriba();
  moverArriba();
  moverArriba();
  moverArriba();
  moverArriba();
  moverArriba();
  moverArriba();

  girarIzquierda();
  moverIzquierda();
  moverIzquierda();
  moverIzquierda();
  moverIzquierda();
  moverIzquierda();
  moverIzquierda();
  moverIzquierda();
  moverIzquierda();
  moverIzquierda();

  girarArriba();

  moverArriba();
  moverArriba();
  moverArriba();
}

moverALechuga();
recolectar();

/* */
finalizar(this);


